sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -163,6 +163,30 @@ def test_precomputed(random_state=42):
         assert_array_almost_equal(pred_X, pred_D)
 
 
+def test_precomputed_sparse():
+    dist = np.array([[1., 2., 0.], [5., .5, 0.], [1., 1., .5]])
```

Comment:
This is not a believable distance matrix, as it is not symmetric. This is not really a big deal: it's fine, perhaps, to test that it works how you expect even with unusual input, but either make it symmetric, or comment that you're intentionally testing an extreme case...